collector/fixtures/e2e-64k-page-output.txt

```diff
@@ -1790,6 +1799,9 @@ node_memory_KernelStack_bytes 5.9392e+06
 # HELP node_memory_Mapped_bytes Memory information field Mapped_bytes.
 # TYPE node_memory_Mapped_bytes gauge
 node_memory_Mapped_bytes 2.4496128e+08
+# HELP node_memory_MemAvailable_bytes Memory information field MemAvailable_bytes.
+# TYPE node_memory_MemAvailable_bytes gauge
+node_memory_MemAvailable_bytes 0
```

Comment:
Yes, exactly like this problem. Not all versions of Linux support `MemAvailable`. So we have PromQL fallback logic to handle situations like this.

```
node_memory_MemAvailable_bytes
or
( 
  node_memory_Buffers_bytes +
  node_memory_Cached_bytes +
  node_memory_MemFree_bytes +
  node_memory_Slab_bytes
) 
```

Turning nil into 0 will break older systems.

Reply:
Addressed in updated commits :+1: